include/deal.II/base/tensor_function_parser.h

```diff
@@ -0,0 +1,361 @@
+ *   argument. If this kind of behavior is a problem, you can resort to
+ *   the muparser built-in syntax `(condition ? then-value : else-value)`,
+ *   using the ternary syntax familiar to C++ programmers. If this
+ *   syntax is used, muparser uses lazy evaluation in which only one of the
+ *   branches is evaluated, depending on whether the `condition` is
+ *   true or false.
+ *
+ * This class overloads the virtual method value() and value_list() of the
+ * TensorFunction base class with the byte compiled versions of the expressions
+ * given to the initialize() methods. Note that the class will not work unless
+ * you first call the initialize() method that accepts the text description of
+ * the function as an argument (among other things).
+ *
+ * The syntax to describe a function follows usual programming practice, and
+ * is explained in detail at the homepage of the underlying muparser library
+ * at http://muparser.beltoforion.de/ .
+ *
+ *
+ * Vector-valued functions can either be declared using strings where the
+ * function components are separated by semicolons, or using a vector of
+ * strings each defining one vector component.
+ *
+ *
+ * @ingroup functions
+ * @author Konrad Simon, 2019
+ */
+template <int rank, int dim, typename Number = double>
+class TensorFunctionParser : public TensorFunction<rank, dim, Number>
+{
+public:
+  /**
+   * Standard constructor. Only set initial time. This object needs to be initialized
+   * with the initialize() method before you can use it. If an attempt to
+   * use this function is made before the initialize() method has been called,
+   * then an exception is thrown.
+   */
+  TensorFunctionParser(const double initial_time = 0.0);
+
+  /**
+   * Constructor for parsed functions. This object needs to be initialized
+   * with the initialize() method before you can use it. If an attempt to
+   * use this function is made before the initialize() method has been called,
+   * then an exception is thrown.
+   * Takes a semi-colon separated list of expressions (one for each component
+   * of the tensor function), an optional comma-separated list of constants.
+   */
+  TensorFunctionParser(const std::string &expression,
+                 const std::string &constants      = "",
+                 const std::string &variable_names = default_variable_names() +
+                                                     ",t");
+
+  /**
+   * Destructor. Explicitly delete the TensorFunctionParser objects (there is one
+   * for each component of the function).
+   */
+  ~TensorFunctionParser() override;
+
+  /**
+   * Type for the constant map. Used by the initialize() method.
+   */
+  using ConstMap = std::map<std::string, double>;
+
+  /**
+   * Iterator for the constants map. Used by the initialize() method.
+   */
+  using ConstMapIterator = ConstMap::iterator;
+
+  /**
+   * Initialize the tensor function.  This methods accepts the following parameters:
```

Comment:
```suggestion
   * Initialize the tensor function.  This constructor accepts the following parameters:
```

Reply:
Since the documentation here is the same as that of FunctionParser it would be better to write something like "this has the same arguments as FunctionParser::initialize()" instead of copying and pasting.